azkaban-webserver/src/main/java/azkaban/webapp/servlet/ExecutorServlet.java

```diff
@@ -338,6 +340,24 @@ private void handleExecutionFlowPage(HttpServletRequest req,
       page.render();
       return;
     }
+    
+    Props props = getApplication().getServerProps(); 
+    String execExternalLinkURL = 
+        props.getString("execution.external.link.url", "");
+    String execExternalLinkLabel = 
+        props.getString("execution.external.link.label", "External Analyzer");
+        
+    if (StringUtils.isNotEmpty(execExternalLinkURL)) {
+      StringBuilder builder = new StringBuilder();
+      builder.append(req.getRequestURL());
+      builder.append("?");
+      builder.append(req.getQueryString());
+      String flowExecutionURL = builder.toString();
```

Comment:
I discussed this with Shankar and found that a given execution ID is not unique across azkaban instances and hence they use the hostname in the url to distinguish between two flow executions. For simplicity, they used the URL. The portname itself is not required though.

He had another suggestion where-in we replace a pattern in the url instead of '%s'

**Sample code**:
`String link= "http://elephant.linkedin.com:8080/search?flow-exec-id=%url";`	  	  
`String pattern = link.substring(link.indexOf('%'), link.length());`	  
`	    switch (pattern) {`
`	      case "%flowExecId": System.out.println(link.replaceFirst(pattern, "100")); break;`
`             case "%url": System.out.println(link.replaceFirst(pattern, "200")); break;`
`	      default: System.out.println("Pattern not found."); break;`
`	  }`
`  }`

Please let me know if you are fine with this approach.